### ISSUE TITLE
DPDHL: Wrong use of street number

### ISSUE THREAD

**Issue by MarcoWel** [karrioapi/karrio#344]
DPDHL API expects streetName (name of street without house number) and streetNumber (house number, up to 10 digits) parameters. Alternatively, street name incl. street number can passed in streetName, but in that case streetNumber **must** be ommitted.

Karrio passes Street Line 1 for streetName and Street Line 2 for streetNumber which is wrong:
```
                        <Address>
                            <cis:streetName>Teststr. 123</cis:streetName>
                            <cis:streetNumber>Additional Address Line</cis:streetNumber>
```

Correct would be:
```
                        <Address>
                            <cis:streetName>Teststr. 123</cis:streetName>
                            <cis:addressAddition>Additional Address Line</cis:addressAddition>
```

If streetNumber is used, Karrio would need to identify the street number and split it automatically (e.g. via regex).
Please note that for some countries the street number comes before the street name (France, US, ...) and for some after the street name (Germany, Netherlands, ...).

DHL API Docs describe all allowed address fields in: 2.2.4.2.	ShipmentOrder.Shipment.Receiver.Address
https://entwickler.dhl.de/group/ep/wsapis/geschaeftskundenversand/current/operationen/createshipmentorder/io-referenz

**Comment by danh91** [comment]
It is confusing indeed.
We've realized that we need an actual explicit `street_name` property because we had to deal with similar situations with other carriers.

We will probably make the street name an explicit property of the Address model. (regex for this would be opening a can of worms 😅)

**Comment by danh91** [comment]
> A dedicated `street_name` property including the street number is fine to us, however, I would opt against a dedicated `street_number` property as most systems do not offer this.

I made a typo. I actually mean `street_number` because it will be optional and when provided, it can be concatenated with the `address_line1` for carriers that do not need them separated and passed directly for carrier APIs that support it.

**Comment by MarcoWel** [comment]
Seems like this issue has not been solved or reintroduced with 2023.4. The field "Street No." is generally omitted for DPDHL.

The field "addressAddition" suggested earlier in this issue here should actually also not be used though:

```
ShipmentOrder.Shipment.Receiver.Address.addressAddition
String 	maxLength: 35	Optional
Address addon, is only printed in the international area (V53WPAK)
```


For the correct mapping, please see new related issue #370.
